components/Variables/context.tsx

```diff
@@ -0,0 +1,66 @@
+import { createContext, useCallback, useMemo, useState } from "react";
+
+export interface VarsContextProps {
+  fields: {
+    [name: string]: string;
+  };
+  globalFields: {
+    [name: string]: boolean;
+  };
+  setField: (name: string, value: string) => void;
+  addField: (name: string) => void;
+  markGlobalField: (name: string) => void;
+}
+
+export const VarsContext = createContext<VarsContextProps>({
+  fields: {},
+  globalFields: {},
+  setField: () => undefined,
+  addField: () => undefined,
+  markGlobalField: () => undefined,
```

Comment:
We can't do it in syntax that you suggested. Because this syntax can be only used in the interface. Here we have default values.

These fields are crucial for context work. I am not sure that these fields should be optional.

Reply:
Ah, yes, I thought I was looking at a type.